otherlibs/systhreads/st_posix.h

```diff
@@ -90,8 +91,14 @@ static void st_thread_join(st_thread_id thr)
 
 static void INLINE st_thread_yield(void)
 {
-#ifndef __linux__
+#ifdef __linux__
   /* sched_yield() doesn't do what we want in Linux 2.6 and up (PR#2663) */
+  /* but not doing anything here would actually disable preemption (PR#7669) */
+  struct timespec t;
+  t.tv_sec = 0;
+  t.tv_nsec = 1;
+  nanosleep(&t, &t);
+#else
```

Comment:
It is nicer to use `NULL` as second argument of `nanosleep` since it makes it obvious that you don't need the remaining time to be stored in `t`.

Reply:
Done.